cachito/workers/pkg_managers/gomod.py

```diff
@@ -154,19 +165,22 @@ def resolve_gomod(app_source_path, request, dep_replacements=None):
         )
         module = {"name": module_name, "type": "gomod", "version": module_version}
 
-        bundle_dir = RequestBundleDir(request["id"])
+        if "gomod-vendor" not in request.get("flags", {}):
+            bundle_dir = RequestBundleDir(request["id"])
 
-        # Add the gomod cache to the bundle the user will later download
-        tmp_download_cache_dir = os.path.join(temp_dir, RequestBundleDir.go_mod_cache_download_part)
-        if not os.path.exists(tmp_download_cache_dir):
-            os.makedirs(tmp_download_cache_dir, exist_ok=True)
+            # Add the gomod cache to the bundle the user will later download
+            tmp_download_cache_dir = os.path.join(
+                temp_dir, RequestBundleDir.go_mod_cache_download_part
+            )
+            if not os.path.exists(tmp_download_cache_dir):
+                os.makedirs(tmp_download_cache_dir, exist_ok=True)
 
-        log.debug(
-            "Adding dependencies from %s to %s",
-            tmp_download_cache_dir,
-            bundle_dir.gomod_download_dir,
-        )
-        shutil.copytree(tmp_download_cache_dir, str(bundle_dir.gomod_download_dir))
+            log.debug(
+                "Adding dependencies from %s to %s",
+                tmp_download_cache_dir,
+                bundle_dir.gomod_download_dir,
+            )
+            shutil.copytree(tmp_download_cache_dir, str(bundle_dir.gomod_download_dir))
```

Comment:
When the flag is provided, could you please create an empty directory at `bundle_dir.gomod_download_dir`? This would make sure that any consumer of Cachito that relies on that directory being present in the bundle would still work.